lib/validation/schema/specVersion/kind/project.json

```diff
@@ -381,6 +396,120 @@
 				}
 			}
 		},
+		"builder-bundles-bundleDefinition-4.0": {
+			"type": "object",
+			"additionalProperties": false,
+			"required": ["name"],
+			"properties": {
+				"name": {
+					"type": "string"
+				},
+				"defaultFileTypes": {
+					"type": "array",
+					"items": {
+						"type": "string"
+					}
+				},
+				"sections": {
+					"type": "array",
+					"if": {
+						"$comment": "Add async prop only if mode = 'require'",
```

Comment:
this constraint does not work. I suspect that the "then" section must not list all the other modes and only list "require" (see comment below).

It looks like there's a test missing to verify that the async flag is not allowed in other section modes.

Reply:
Fixed with: https://github.com/SAP/ui5-project/pull/751